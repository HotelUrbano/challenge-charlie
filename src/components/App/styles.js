import styled from "styled-components";

export const App = styled.main`
  background: url(${(props) => props.backgroundUrl});
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  display: flex;
  height: 100vh;
  justify-content: center;
`;