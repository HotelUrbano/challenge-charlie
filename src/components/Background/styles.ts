import styled from 'styled-components';

interface ContainerProps {
  imageUrl?: string;
}

export const Container = styled.div<ContainerProps>`
  background: url(${(props) => `https://www.bing.com/${props.imageUrl}`})
    no-repeat center;
  background-size: cover;
  height: 100vh;
  overflow-y: scroll;

  ::-webkit-scrollbar {
    display: none;
  }
`;
