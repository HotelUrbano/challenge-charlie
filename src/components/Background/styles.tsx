import styled from 'styled-components';

interface BackgroundProps {
    bg?: any;
}
export const Container = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;

    background-image: ${(props: BackgroundProps) =>
        props.bg ? `url(https://www.bing.com${props.bg})` : '#fffff'};

    width: 100%;
    height: 100vh;
`;
