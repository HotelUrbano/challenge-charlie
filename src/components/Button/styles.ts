import styled from 'styled-components';
import { shade } from 'polished';

export const Container = styled.button`
    background: var(--hu-color);
    border-radius: 10px;
    border: 0;
    padding: 10px;
    height: 100%;
    width: 200px;
    color: var(--white-color);
    font-weight: 500;
    margin-left: 8px;
    transition: background .2s;

    &:hover {
        background: ${shade(.2, '#083D7D')};
    }
`;

export const Loading = styled.div`
    display: inline-block;
    width: 100%;
    height: 100%;

    display: flex;
    align-items: center;

    &:after {
        content: "";
        display: block;
        width: 16px;
        height: 16px;
        margin: auto;
        border-radius: 50%;
        border: 3px solid var(--white-color);
        border-color: var(--white-color) transparent var(--white-color) transparent;
        animation: spinner 1.2s linear infinite;
    }

    @keyframes spinner {
        0% {
            transform: rotate(0deg);
        }
        100% {
            transform: rotate(360deg);
        }
    }
`;