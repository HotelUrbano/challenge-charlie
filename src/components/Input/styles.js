import styled from 'styled-components';

export const Container = styled.div`
  width: 100%;
  display: flex;
  background: #fff;
  padding: 5px;

  svg {
    width: 64px;
    height: 64px;
    path {
      fill: #8c8885;
    }
  }

  input {
    height: 64px;
    width: 100%;
    margin-left: 15px;
    border: none;
    font-size: 1.8em;
    color: #8c8885;
    background: inherit;
  }
`;
