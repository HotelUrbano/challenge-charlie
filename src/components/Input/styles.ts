import styled, { css } from 'styled-components';

interface ContainerProps {
    isFocused: boolean;
    isFilled: boolean;
}

export const Container = styled.div<ContainerProps>`
    background-color: #E1E1E1;
    color: var(--black-color);
    border-radius: 8px;
    padding: 10px;
    width: 100%;
    display: flex;
    align-items: center;
    border: 2px solid #E1E1E1;

    ${props => props.isFocused && css`
        border-color: var(--hu-color);
        color: var(--hu-color);
    `}

    ${props => props.isFilled && css`
        color: var(--hu-color);
    `}

    input {
        color: var(--black-color);
        background: transparent;
        flex: 1;
        width: 100%;
        border: 0;

        &::placeholder {
            color: #555;
        }
    }

    svg {
          margin-right: 16px;
    }
`;