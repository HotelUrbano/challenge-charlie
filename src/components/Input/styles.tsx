import styled from 'styled-components';

export const Container = styled.div`
    display: flex;
    width: 100%;
    border: 1px solid #ccc;
    border-radius: 4px;
    margin-right: 0.5rem;
    padding: 0.5rem;
    > input {
        width: 100%;
    }
`;
