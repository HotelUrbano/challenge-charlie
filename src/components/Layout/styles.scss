$font-size: 1rem;
$line-height-base: 1.5;

html {
  @media (max-width: 724px) {
    font-size: 14px;
  }

  .layout {
    max-width: 100vw;

    > .background {
      position: absolute;
      z-index: -1;

      width: 100%;
      max-width: 100%;
      height: 100%;

      background-image: url(https://www.bing.com/th?id=OHR.BrightEye_ROW7827840218_1920x1080.jpg&rf=LaDigue_1920x1080.jpg&pid=hp);
      filter: brightness(30%);
    }

    p, small {
      color: $grey;

      font-weight: 500;
      font-family: 'Noto Sans JP', sans-serif;

      font-size: $font-size;
      line-height: $line-height-base;
    }
  }
}

