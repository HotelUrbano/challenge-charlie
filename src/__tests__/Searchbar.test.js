import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import SearchBar from '../components/SearchBar/SearchBar';
import React from 'react';
import { SearchBarInput } from '../components/SearchBar/SearchBar.styles';


const setCityMock = jest.fn()

const SearchBarComponentProps = {
	address: {},
	city: 'Cidade123',
	setCity: setCityMock,
	searchCity: () => true,
	loading: false
}

describe('SearchBar Component', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(
			<SearchBar {...SearchBarComponentProps}>
			</SearchBar>
		);
	});

	it('Renders', () => {
		const SearchBarComponent = renderer.create(<SearchBar {...SearchBarComponentProps}></SearchBar>);
		expect(SearchBarComponent).toMatchSnapshot();
	});

	it('Sets input value to city variable', () => {

		const event = {
			target: {
				value: 'teste123'
			}
		}

		wrapper.find(SearchBarInput).simulate('change', event)
		expect(setCityMock).toHaveBeenCalledWith('teste123')
	})
});