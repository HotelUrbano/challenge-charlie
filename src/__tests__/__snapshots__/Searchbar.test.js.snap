// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Component Renders 1`] = `
<section
  className="sc-bdfBwQ iIzYUp"
>
  <img
    className="sc-hKgILt eBBmrW"
    src="public/img/44.svg"
  />
  <form
    className="sc-gsTCUz cVyLar"
    onSubmit={[Function]}
  >
    <input
      className="sc-dlfnbm gHrCKK"
      onChange={[Function]}
      placeholder="Type your location"
      type="text"
      value="Cidade123"
    />
  </form>
</section>
`;
